src/test/java/com/synopsys/integration/detect/battery/tests/GradleBattery.java

```diff
@@ -0,0 +1,42 @@
+package com.synopsys.integration.detect.battery.tests;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.detect.battery.BatteryTest;
+import com.synopsys.integration.detect.battery.ResourceCopyingExecutableCreator;
+import com.synopsys.integration.detect.configuration.DetectProperty;
+
+@Tag("battery")
+public class GradleBattery {
+    private static final String RESOURCE_FOLDER = "GRADLE-0";
+
+    @Test
+    void property() {
+        final BatteryTest test = sharedInspectorTest("gradle-property");
+        addGradleArguments(test.executableThatCopiesFiles(DetectProperty.DETECT_GRADLE_PATH, RESOURCE_FOLDER));
+        test.run();
+    }
+
+    @Test
+    void wrapper() {
+        final BatteryTest test = sharedInspectorTest("gradle-wrapper");
+        addGradleArguments(test.executableSourceFileThatCopiesFiles("gradlew.bat", "gradlew", RESOURCE_FOLDER));
+        test.run();
+    }
+
+    //Note about this test: The paths have been removed from the inspector meta data.
+    BatteryTest sharedInspectorTest(final String name) {
+        final BatteryTest test = new BatteryTest(name, "gradle-inspector");
+        test.sourceDirectoryNamed("linux-gradle");
+        test.sourceFileNamed("build.gradle");
+        test.git("https://github.com/BlackDuckCoPilot/example-gradle-travis", "HEAD");
+        test.expectBdioResources();
+        return test;
+    }
```

Comment:
I think using our own open sourced repos like we did here is better to using other projects.

Reply:
I mean we don't actually use any of the source code - maybe lock files could be considered source? But using repo's we own could be preferred.

Reply:
I would consider lock files part of source. But if we are okay with using open source projects in our tests then that's fine.